### ISSUE TITLE
Fails to download libWiiSharp.dll and doesn't patch the WADs

### ISSUE THREAD

**Issue by adueppen** [RiiConnect24/RiiConnect24-Patcher#11]
I'm attempting to use the patcher on Linux (Fedora 28) and at around 30% the patcher displays this message:
```WARNING: The runtime version supported by this application is unavailable.
Using default runtime: v4.0.30319
ERROR: libWiiSharp.dll not found


Attemp to download? [Y/N]

>>
```
Selecting either yes or no will advance the progress bar and eventually cause the patcher to print its success message, but there is nothing in the WAD folder. When selecting yes, something about an SSL trust error appears, but disappears too quickly for me to copy and paste here.

**Comment by Artuto** [comment]
What happens if you try to download the file [directly](https://raw.githubusercontent.com/KcrPL/KcrPL.github.io/master/Patchers_Auto_Update/RiiConnect24Patcher/IOSPatcher/libWiiSharp.dll)?

**Comment by adueppen** [comment]
The file is there in the folder but doesn't seem to be recognized.
![screenshot-20180809195557-782x654](https://user-images.githubusercontent.com/6982070/43933251-423389f8-9c0e-11e8-9751-dca80a13fbe6.png)

**Comment by ApfelTV** [comment]
i will check the code asap, hopefully its just a url typo.

**Comment by adueppen** [comment]
The "Attemp to download" message seems to be something in Sharpii.exe since it's nowhere to be found in the script.

**Comment by ApfelTV** [comment]
Yes, Sharpii checks if a file doesnt exists and ask the user to download it. i will check the url now.

**Comment by ApfelTV** [comment]
and could you check if the `libWiiSharp.dll` exist in the `EVCPatcher` directory?

**Comment by adueppen** [comment]
It doesn't appear to

**Comment by ApfelTV** [comment]
[download the dll here](https://raw.githubusercontent.com/KcrPL/KcrPL.github.io/master/Patchers_Auto_Update/RiiConnect24Patcher/EVCPatcher/libWiiSharp.dll), and place it in the `EVCPatcher` directory.

**Comment by adueppen** [comment]
I tried doing that and the patcher seemingly made it a bit farther than before, including doing the cleanup at the end, but the final WAD folder is still empty. At the very least, it only asked once about the dll during the EVC patching section.

**Comment by adueppen** [comment]
I just tried doing the manual install option and just patching the IOSs and that seemed to work, but do you have the MD5s of the patched IOSs to check?

**Comment by ApfelTV** [comment]
sadly no.

**Comment by ApfelTV** [comment]
but wait i can check them

**Comment by adueppen** [comment]
The EVC patching menu in the manual install section also seemed to work, so I'm not sure what the problem is.

**Comment by adueppen** [comment]
I was able to get the error message from Sharpii.exe and added it to the original report. It would seem that it/Mono might not have the DigiCert Root CA.

**Comment by adueppen** [comment]
I'm not sure how useful this is, but I also found out that part of the reason for the patcher failing is on lines 549 and 552. When using Sharpii.exe on its own with Wine and running the script line by line, those lines fail because there is no wad file called `000000010000001fv3608.wad` (for example). There is one named `IOS31-64-3608.wad` (in the case of IOS31) but patching it with xdelta still produces a file with a different hash.

**Comment by dotcomboom** [comment]
Works for me on Ubuntu after adding the Mono PPA/repo and upgrading my packages.

**Comment by adueppen** [comment]
OK so I tried updating mono and it's still giving me the unsupported runtime warning, but at the very least it's successfully downloading libWiiSharp.dll now.

**Comment by ApfelTV** [comment]
i will later build a .net 4.7.1 version, hopefully that will fix that.

**Comment by ApfelTV** [comment]
nvm it was an url issue, it got fixed
this issue can be closed now
